blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/BlueTestResultContainerImpl.java

```diff
@@ -63,62 +63,93 @@ public BlueTestResult get(final String name) {
             throw new NotFoundException("no tests");
         }
         StaplerRequest request = Stapler.getCurrentRequest();
-        Iterator<BlueTestResult> results;
         if (request != null) {
             String status = request.getParameter("status");
             String state = request.getParameter("state");
-            if (isEmpty(status) && isEmpty(state)) {
-                results = resolved.results.iterator();
-            } else if (!isEmpty(status)) {
-                results = filterByStatus(resolved.results, status);
-            } else if (!isEmpty(state)) {
-                results = filterByState(resolved.results, state);
-            } else {
-                throw new BadRequestException("must provide either 'status' or 'state' params");
-            }
-        } else {
-            results = resolved.results.iterator();
+            String age = request.getParameter("age");
+            return getBlueTestResultIterator(resolved.results, status, state, age);
+
+        }
+        return resolved.results.iterator();
+    }
+
+    @VisibleForTesting
+    public Iterator<BlueTestResult> getBlueTestResultIterator(Iterable<BlueTestResult> results, String status, String state, String age) {
```

Comment:
You've updated the backend to gracefully handle missing search params here it seems, probably can remove the exception at the front-end if they're missing, also.

Reply:
Agreed, done